tap_recharge/client.py

```diff
@@ -101,20 +102,34 @@ class RechargeClient:
     def __init__(
             self,
             access_token,
-            user_agent=None):
+            user_agent=None,
+            request_timeout=REQUEST_TIMEOUT):
         self.__access_token = access_token
         self.__user_agent = user_agent
         self.__session = requests.Session()
         self.__base_url = None
         self.__verified = False
-
+        # if request_timeout is other than 0,"0" or "" then use request_timeout
+        if request_timeout and float(request_timeout):
+            request_timeout = float(request_timeout)
+        else: # If value is 0,"0" or "" then set default to 300 seconds.
+            request_timeout = REQUEST_TIMEOUT
+        self.request_timeout = request_timeout
+
+    # Backoff the request for 5 times when Timeout or Connection error occurs
+    @backoff.on_exception(
+        backoff.expo,
+        (Timeout, requests.ConnectionError),
+        max_tries=5,
+        factor=2)
```

Comment:
@namrata270998 Can we merge the above backoff with the already available backoff on `check_access_token` function for consistency?
https://github.com/singer-io/tap-recharge/blob/acbcb56825ad8fc64d62f0786a51b100f5e753e1/tap_recharge/client.py#L133-L138

Reply:
We cannot merge the 2 backoffs as the `check_access_token()` is also called from the `request()` function which has a backoff. Hence, to avoid more than 5 times backoff, it is places on the top of __enter__()